### PR TITLE
BELL-4737: Misha's first ticket

### PR DESCRIPTION
**Pull request by MishaShapo**
- DropWizard app that supports 2 types of calculators: a basic 4 function and a scientific calculator
- Added dependency injection for the CalculatorResource and the health check with guice
- Basic unit testing with DropWizard Health Checks

### CODE REVIEW

File: src/main/java/com/shaposhnikov/michail/core/CalculatorResource.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        this.counter = new AtomicLong();
+        this.errors = new ArrayList<>();
+        this.calculatorService = calculatorService;
+    }
+
+    @GET
+    @Path ("{operation: (add|sub|mul|div|sqrt|pow)}")
+    @Timed
+    public ArithmeticResult operation(@QueryParam ("val") List<String> values, @PathParam ("operation") String operationParam) {
+        this.errors.clear();
```
Comment (DarthLorenzo) [inline src/main/java/com/shaposhnikov/michail/core/CalculatorResource.java]: This is very close to working. If you are the only one using this webservice, then the application will only be handling one request at a time. In general, Dropwizard will make multiple threads use this Resource. Having the errors of an API call stored in this class variable means that the errors of one thread could be cleared out from a subsequent thread.

TL;DR: See if you can re-organize the code to store errors differently.

**Review by DarthLorenzo — CHANGES_REQUESTED** [review]

File: src/main/java/com/shaposhnikov/michail/core/CalculatorResource.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            return calculatorService.applyOperationTo(this.counter.getAndIncrement(),numbers,operationParam);
+        }
+        return ArithmeticResult.generateErroredResult(this.counter.getAndIncrement(),this.errors);
+    }
+
+    private List<Double> verify(List<String> values){
+        List<Double> numbers = new ArrayList<>();
+        try{
+            for (String value: values){
+                double number = Double.parseDouble(value);
```
Comment (DarthLorenzo) [inline src/main/java/com/shaposhnikov/michail/core/CalculatorResource.java]: This would be a great opportunity for a simple stream.

File: src/main/java/com/shaposhnikov/michail/core/CalculatorResource.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @GET
+    @Path ("{operation: (add|sub|mul|div|sqrt|pow)}")
+    @Timed
+    public ArithmeticResult operation(@QueryParam ("val") List<String> values, @PathParam ("operation") String operationParam) {
+        this.errors.clear();
+
+        List<Double> numbers = verify(values);
+        if(this.errors.size() == 0 ){
```
Comment (DarthLorenzo) [inline src/main/java/com/shaposhnikov/michail/core/CalculatorResource.java]: List.isEmpty() is a great method you've used other places
